src/content/docs/query-your-data/nrql-new-relic-query-language/nrql-query-tutorials/NRQL-high-cardinality-metrics.mdx

```diff
@@ -0,0 +1,123 @@
+
+## Cardinality and dimensional metrics 
+
+For dimensional metrics, the set in question is the collection of observed unique combinations of attributes for a given metric in a one-day period. Each group of attributes is a mapping of key-value that's counted as a single element of the set. 
+
+Imagine a network of 4 hosts, each with 2 containers running on them, and each container periodically reports the gauge metric `memory.heap`, with the host name and container id added as attributes. 
+
+![Host and containers combination resulting in high cardinality](./images/high-cardinality-metrics-1.png "Multiple hosts with multiple containers")
+
+When submitted to the Metric API, one of these metrics might look something like this:
+
+```
+
+"metrics":[
+  { 
+"name":"memory.heap", 
+"type":"gauge", 
+"value":5514, 
+"timestamp":1234567890, 
+"attributes":{
+"host":"W",
+"container":"1"
+    }
+  }
+]
+```
+
+This metric has a cardinality of 8, with each container counted as a single element in the set. If a new measurement for this metric with an identical attributes map as one that had previously arrived later, no new cardinality would be counted.
+
+## Cardinality influences
+
+Any changes to the keys or values will represent new cardinality, but predicting how those changes will impact your total cardinality can get a little tricky. 
+
+It's' tempting to assume then that the cardinality of a metric is the product of the number of all possible values for each possible key, but this is rarely the case in practice. This is because the values a given key can have are often dependent on the values of other keys. 
+
+For example, given a `container.id`, the possible value of `host.name` is likely determined assuming those IDs are globally unique. So while you may have 100 containers across 10 hosts, the cardinality is still 100, not 10 * 100 = 1000, which is what  the product method for determining cardinality would yield. Put another way, it's often true that most combinations counted in this way are not possible and therefore never contribute to that metric's cardinality. 
+
+This also means that adding more keys to an attributes map does not imply an increase to total cardinality if the value of the new key is uniquely determined by the values of existing keys. For instance, if you add something like region to your map, it's likely the case that the `container.id` is fixed to a particular region value. 
+
+An important caveat here is that while adding region won't increase the cardinality going forward, it will introduce new cardinality when it is first added. Also, recall that adding keys will make those attributes' maps distinct from any that came before them, temporarily increasing the total cardinality for that day.
+
+## Examples and sample workflows
+
```

Comment:
(Is this a good place to discuss how to query for cardinality violations (if so here is a suggestion)

If you believe you have hit one of your cardinality limits, you can confirm this by checking for a related **NrIntegrationError**:

```
FROM NrIntegrationError SELECT count(*) where name = 'CardinalityViolationException' and newRelicFeature = 'Metrics' facet cardinalityLimitType, metricName, message since today 
```

Reply:
Okay, I've added this into the metrics section @jonahaapala - you can take a look once I add this next round of commits.

Reply:
I like the idea of showing how to query for the presence of the problem before diving into exploration and mitigation, so I'm +1 on adding that here. We should remove it from 105-109 below so it's not duplicated.

Reply:
Yeah I think it makes sense to add here as well